### PR TITLE
fix: data submission policy version

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
## Reason for Change

- #6202

## Changes

- Updated publish of collection to save data submission policy version to DB.
- Updated read of collection to return saved data submission policy version. 

## Testing steps

- Updated controller and business-layer tests.
- Tested in rdev.
 
## Checklist 🛎️

- [ ] Add product, design, and eng as reviewers for rdev review

### CODE REVIEW

**Review by nayib-jose-gloria — COMMENTED** [review]
looks good, thank you! one comment about gating against input submissions policies < our current submission policy

**Review by ebezzi — APPROVED** [review]

File: backend/portal/api/portal_api.py [hunk 1]
```diff
@@ -68,18 +70,18 @@ def get_collections_list(from_date: int = None, to_date: int = None, token_info:
     user_private_collections = get_user_private_collections(token_info)
 
     collections = []
-    for version in itertools.chain(all_published_collections, user_private_collections):
+    for collection_version in itertools.chain(all_published_collections, user_private_collections):
```
Comment (MillenniumFalconMechanic) [inline backend/portal/api/portal_api.py:73]: `version` here shadowed imported `version`; updated to `collection_version`.

File: backend/portal/api/portal_api.py [hunk 2]
```diff
@@ -5,7 +5,9 @@
 from urllib.parse import urlparse
 
 from flask import Response, jsonify, make_response
+from packaging import version
```
Comment (ebezzi) [inline backend/portal/api/portal_api.py:8]: I'm not sure introducing a dependency for something this simple is a good idea. Any reason for which you can't just compare the two strings (or even two parsed numbers)?

Comment (MillenniumFalconMechanic) [reply backend/portal/api/portal_api.py:8]: The original requirement was `>=` but I think now it makes sense to just do a string comparison.

**Review by ebezzi — APPROVED** [review]

**Pull request merged**